src/module.cpp

```diff
@@ -2285,6 +2290,7 @@ void Module::execPreprocessor(const char *infilename, llvm::raw_string_ostream *
 
     llvm::IntrusiveRefCntPtr<clang::DiagnosticIDs> diagIDs(new clang::DiagnosticIDs);
     clang::DiagnosticsEngine *diagEngine = new clang::DiagnosticsEngine(diagIDs, diagOptions, diagPrinter);
```

Comment:
For my understanding: Do we care about the leaks in the preprocessor? I think this leaks for every module processed.

Reply:
In general yes, we care about leaks, we should avoid all the leaks. But there are things that fundamentally difficult to fix wrt leaks, so we didn't pay close attention to it.